src/aind_data_transfer/jobs/basic_job.py

```diff
@@ -0,0 +1,201 @@
+            s3_client.close()
+        if results["KeyCount"] != 0:
+            raise FileExistsError(
+                f"S3 path s3://{self.configs.s3_bucket}/"
+                f"{self.configs.s3_prefix} already exists. Please consult "
+                f"your admin for help removing old folder if desired."
+            )
+        return None
+
+    def _compress_raw_data(self, temp_dir):
+
+        if self.configs.behavior_dir is not None:
+            behavior_dir_excluded = self.configs.behavior_dir / "*"
+        else:
+            behavior_dir_excluded = None
+
+        # If no compression is required, we'll upload the data directly instead
+        # of copying it to a temp folder
+        if not self.configs.compress_raw_data:
+            # Upload non-behavior data to s3
+            data_prefix = "/".join(
+                [self.configs.s3_prefix, self.configs.experiment_type.value]
+            )
+            upload_to_s3(
+                directory_to_upload=self.configs.data_source,
+                s3_bucket=self.configs.s3_bucket,
+                s3_prefix=data_prefix,
+                dryrun=self.configs.dry_run,
+                excluded=behavior_dir_excluded,
+            )
+        # Otherwise, we'll store the compressed folder in a temp directory
+        # and upload it along with the other files
+        else:
+            logging.info("Compressing raw data folder: ")
+            zc = ZipCompressor(display_progress_bar=True)
+            compressed_data_folder_name = (
+                str(os.path.basename(self.configs.data_source)) + ".zip"
+            )
+            folder_path = (
+                temp_dir
+                / self.configs.experiment_type.value
+                / compressed_data_folder_name
+            )
+            os.mkdir(folder_path.parent)
+            skip_dirs = (
+                None
+                if self.configs.behavior_dir is None
+                else [self.configs.behavior_dir]
+            )
+            zc.compress_dir(
+                input_dir=self.configs.data_source,
+                output_dir=folder_path,
+                skip_dirs=skip_dirs,
+            )
+        return None
+
+    def _compile_metadata(self, temp_dir):
+
+        # Get metadata from service
+        subject_metadata = SubjectMetadata.from_service(
+            domain=self.configs.metadata_service_domain,
+            subject_id=self.configs.subject_id,
+        )
+        subject_file_name = temp_dir / subject_metadata.output_filename
+        subject_metadata.write_to_json(subject_file_name)
+        procedures_metadata = ProceduresMetadata.from_service(
+            domain=self.configs.metadata_service_domain,
+            subject_id=self.configs.subject_id,
+        )
+        procedures_file_name = temp_dir / procedures_metadata.output_filename
+        procedures_metadata.write_to_json(procedures_file_name)
+        data_description_metadata = RawDataDescriptionMetadata.from_inputs(
```

Comment:
do you need experiment type here?

Reply:
It uses the parse_name method from RawDataDescription (line 300 in aind_data_schema.data_description) to pull exp type from the inputs.